text/0000-associated-type-bounds.md

```diff
@@ -100,6 +102,53 @@ Meanwhile, the surface syntax `dyn Trait<AssociatedType: Bounds>` desugars into
 `dyn Trait<AssociatedType = T>` where `T` is a named type variable `T` with the
 bound `T: Bounds`.
 
+## The desugaring for associated types
+
+In the case of an associated type having a bound of the form:
+
+```rust
+trait TraitA {
+    type AssocA: TraitB<AssocB: TraitC>;
+}
+```
+
+we desugar to an anonymous associated type for `AssocB`, which corresponds to:
+
+```rust
+trait TraitA {
+    type AssocA: TraitB<AssocB = Self::AssocA_0>;
+    type AssocA_0: TraitC; // Associated type is Unnamed!
+}
+```
+
+## Notes on the meaning of `impl Trait<Assoc: Bound>`
+
+Note that in the context `-> impl Trait<Assoc: Bound>`, since the
+`Trait` is existentially quantified, so is in effect also the `Assoc`.
+Semantically speaking, `fn printables..` is equivalent to:
+
+```rust
+fn printables() -> impl Iterator<Item = impl Display> { .. }
+```
+
+For `arg: impl Trait<Assoc: Bound>`, it can likewise be seen as:
+`arg: impl Trait<Assoc = impl Bound>`.
+
+## Meaning of `existential type Foo: Trait<Assoc: Bound>`
```

Comment:
Please include notes in this section that 1) reference the existential type RFC by number and 2) explicitly state that "like that RFC, this documentation uses the non-final syntax for existential type aliases".

Reply:
Also, please edit the tracking issue for that RFC to request that when the syntax gets finalized, this RFC should be updated.